crates/ty_python_semantic/src/types/class_base.rs

```diff
@@ -79,7 +79,14 @@ impl<'db> ClassBase<'db> {
         match ty {
             Type::Dynamic(dynamic) => Some(Self::Dynamic(dynamic)),
             Type::ClassLiteral(literal) => Some(Self::Class(literal.default_specialization(db))),
-            Type::GenericAlias(generic) => Some(Self::Class(ClassType::Generic(generic))),
+
+            Type::GenericAlias(generic) => match generic {
+                GenericAlias::ClassLiteral(generic) => {
+                    Some(Self::Class(ClassType::Generic(generic)))
+                }
+                GenericAlias::TypeAlias(_) => None,
+            },
```

Comment:
You can fold these match statements together like you do below in `display.rs`:

```suggestion
            Type::GenericAlias(GenericAlias::ClassLiteral(generic)) => {
                Some(Self::Class(ClassType::Generic(generic)))
            }
            Type::GenericAlias(GenericAlias::TypeAlias(_)) => None,
```

(Though if you revert `GenericAlias` back to being a struct per above, then you will need to keep this as two separate matches)